src/topology/homotopy/homotopy_group.lean

```diff
@@ -193,3 +355,103 @@ begin
         { convert H.eq_snd _ _, exacts [y.one_char, iH] },
       end }⟩],
 end
+
+section
+variables {n : ℕ} (i : N)
+
+/-- Group structure on `π_(n+1)`. -/
+instance group : group (π_(n+1) X x) :=
+(homotopy_group_equiv_fundamental_group 0).group
+
+/-- Group structure on `homotopy_group` obtained by pulling back path composition along the
+  `i`th direction. The group structures for two different `i j : N` distribute over each
+  other, and therefore are equal by the Eckmann-Hilton argument. When `N = fin (n+1)`,
+  the group structure with `i = 0` is taken to be default and registered as an instance above. -/
+@[reducible] def aux_group (i : N) : group (homotopy_group N X x) :=
+(homotopy_group_equiv_fundamental_group i).group
+
+lemma is_unital_aux_group (i : N) :
+  eckmann_hilton.is_unital (aux_group i).mul (⟦const⟧ : homotopy_group N X x) :=
+⟨⟨(aux_group i).one_mul⟩, ⟨(aux_group i).mul_one⟩⟩
```

Comment:
I think you should be able to prove this with [`eckmann_hilton.mul_one_class.is_unital`](https://leanprover-community.github.io/mathlib_docs/group_theory/eckmann_hilton.html#eckmann_hilton.mul_one_class.is_unital)

Reply:
That second spelling looks good to me.

Reply:
Turns out a type ascription is needed; if you think it's worth it, I'll commit. I don't want to abuse `;` but otherwise it doesn't fit in one line.
```suggestion
by letI : group (homotopy_group N X x) := aux_group i; exact eckmann_hilton.mul_one_class.is_unital
```